src/utils/index.ts

```diff
@@ -33,3 +35,22 @@ export const objectCheck = (data: Data, prefix?: string, postfix?: string): stri
 
   return message;
 };
+
+export class ConfigProvider {
+  private _config: ApiConfig;
+  constructor(config: ApiConfig) {
+    this._config = config;
+  }
+
+  get settings(): ApiConfig {
+    return this._config;
+  }
+
+  set settings(newConfig: ApiConfig) {
+    this._config = newConfig;
+  }
+
+  public setFields(newConfigFields: Partial<ApiConfig>): void {
```

Comment:
Maybe worth returning `this` from here, so that we can chain the calls if needed (`configProvider.setFields().doSomethingElse()` or something similar), but at the same time, we don't really have a use for this at the moment.

Reply:
Made the change